### DISCUSSION TITLE
Why is there getPage("footer") and getExtraHtml(“footer”)?

### DISCUSSION THREAD

**Discussion by Silber-Toaster** [datenstrom/community#982] [Ask a question] [answered]
Hello everyone, 

thank you very much for the good work, 80% of which I can use. I would like to convert the footer to an always present footer... 

I have read the api, but I am missing the explanation of the following variables. how are they filled and why do I need them?

Greetings silver

```
<?php echo $this->yellow->page->getPage("footer")->getContentHtml() ?>
<?php echo $this->yellow->page->getExtraHtml("footer") ?>
```

```
<div class="footer" role="contentinfo">
		<div class="siteinfo">
			<?php echo $this->yellow->page->getPage("footer")->getContentHtml() ?>
			Test NR
		</div>	
		<div class="siteinfo-banner">performance
		</div>
	</div>
</div>
	<?php echo $this->yellow->page->getExtraHtml("footer") ?>
</body>
</html>
```

![Screenshot 2024-07-17 135130](https://github.com/user-attachments/assets/97aa9306-2c0b-4c48-bca8-b5bf8ebb7bf6)

**Comment by schulle4u** [comment, marked as answer]
Welcome on board! 

> ```
> <?php echo $this->yellow->page->getPage("footer")->getContentHtml() ?>
> ```

This API call will include the content from `content/shared/footer.md` as [global page footer](https://datenstrom.se/yellow/help/how-to-make-a-small-website#show-header-and-footer). You can use it for copyright notices, imprints, social media links, privacy policy etc. A similar file `content/shared/header.md` can be created to include additional header content such as a tagline/slogan. 

> ```
> <?php echo $this->yellow->page->getExtraHtml("footer") ?>
> ```

The `getExtraHtml()` call (also available as header and footer) should be mandatory in your layouts as it usually includes additional website ressources defined by extensions (e.g. CSS and JS files or inline code). 

Does this answer your question? 

Happy coding! 🙂

**Comment by Silber-Toaster** [comment]
Thank you very much for the answer, yes my question has been answered.

Greetings Silver
